### PR TITLE
refactor istio to use imagetest_harness_k3s

### PR DESCRIPTION
**Pull request by jamonation**

### CODE REVIEW

**Comment by jamonation** [comment]
This is stuck on creating the svc/istio-system-gateway via the `./test-gateway.sh` script. I'm not sure what the issue is.

**Comment by jamonation** [comment]
Got the gateway test sorted, this is ready for a second set of 👀

File: images/istio/tests/operator/main.tf [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  }
+}
+
+module "helm" {
+  source = "../../../../tflib/imagetest/helm"
+
+  namespace = var.values.namespace
+
+  chart         = "istio-operator"
+  repo          = "https://stevehipwell.github.io/helm-charts/"
```
Comment (joshrwolf) [inline images/istio/tests/operator/main.tf:28]:
```suggestion
  repo          = "https://stevehipwell.github.io/helm-charts/"
```

can we use the official istio helm chart for this? or maybe even `istioctl`?

**Review by joshrwolf — APPROVED** [review]
I realize this is a port of the existing stuff, so no worries about the `stevehipwell` chart in this one.

thanks for doing this!

**Comment by jamonation** [comment]
I looked around a bit for another operator chart, or a way to reuse the official ones but it doesn't seem like there are any with the right knobs and levers, thanks for checking!

**Pull request merged**